presto-orc/src/main/java/com/facebook/presto/orc/OrcWriteValidation.java

```diff
@@ -581,12 +584,12 @@ else if (REAL.equals(type)) {
                 fieldBuilders = ImmutableList.of();
             }
             else if (type instanceof VarcharType) {
-                statisticsBuilder = new StringStatisticsBuilder();
+                statisticsBuilder = new StringStatisticsBuilder().withStringStatisticsLimit(stringStatisticsLimitInBytes);
                 fieldExtractor = ignored -> ImmutableList.of();
                 fieldBuilders = ImmutableList.of();
             }
             else if (type instanceof CharType) {
-                statisticsBuilder = new StringStatisticsBuilder();
+                statisticsBuilder = new StringStatisticsBuilder().withStringStatisticsLimit(stringStatisticsLimitInBytes);
```

Comment:
Maybe require this in the constructor instead of making it optional.  I would hope it make mistakes less likely in the future.